packages/tap-runner/test/integration/tap-test-runner.it.spec.ts

```diff
@@ -115,58 +114,29 @@ describe('tap-runner integration', () => {
       assertions.expectSurvived(run);
     });
 
-    it('should bail out when disableBail is false', async () => {
-      const testFiles = ['tests/bail.spec.js', 'tests/formatter.spec.js'];
-
-      // Act
-      const run = await sut.mutantRun(factory.mutantRunOptions({ testFilter: testFiles, disableBail: false }));
-
-      // Assert
-      assertions.expectKilled(run);
-      expect(run.nrOfTests).eq(1);
-      expect(run.killedBy[0]).eq('tests/bail.spec.js');
-    });
-
-    it('should not bail out when disableBail is true', async () => {
-      const testFiles = ['tests/bail.spec.js', 'tests/formatter.spec.js'];
-
-      // Act
-      const run = await sut.mutantRun(factory.mutantRunOptions({ testFilter: testFiles, disableBail: true }));
-
-      // Assert
-      assertions.expectKilled(run);
-      expect(run.nrOfTests).eq(testFiles.length);
-    });
-
-    it('should bail out current process when disableBail is false and os type is not windows', async function () {
-      if (os.platform() === 'win32') {
-        this.skip();
-      } else {
+    [
+      { disableBail: false, forceBail: false, shouldBail: false },
+      { disableBail: true, forceBail: false, shouldBail: false },
+      { disableBail: true, forceBail: true, shouldBail: false },
+      { disableBail: false, forceBail: true, shouldBail: os.platform() === 'win32' ? false : true },
+    ].forEach(({ disableBail, forceBail, shouldBail }) => {
```

Comment:
Probably better not to generate tests in this case, because you now need a `if` inside the test to figure out what to assert. 

Better would be to create and use helper functions inside designated `it`'s